packages/neon-api/src/provider/notifications/class.ts

```diff
@@ -0,0 +1,48 @@
+import { logging, settings } from "@cityofzion/neon-core";
+import WebSocket from 'isomorphic-ws';
+import { NotificationMessage } from "./responses";
+const log = logging.default("api");
+
+export class Notifications {
+  private url: string;
+
+  public get name(): string {
+    return `Notifications[${this.url}]`;
+  }
+
+  private websocketSubscriptions: Map<string, WebSocket[]>;
+
+  public constructor(url: string) {
+    if (settings.networks[url] && settings.networks[url].extra.notifications) {
+      this.url = settings.networks[url].extra.notifications;
+    } else {
+      this.url = url;
+    }
+	this.websocketSubscriptions = new Map<string, WebSocket[]>();
+    log.info(`Created Notifications Provider: ${this.url}`);
+  }
+
+  public subscribe(contract: string, callback: (message: NotificationMessage) => void) {
+	  const ws = new WebSocket(this.url + "?contract=" + contract);

```

Comment:
new websocket for every callback? We can do better than this. For subscriptions to the same contract, I expect that we reuse the old socket.

Reply:
Agree, working on it.